docs/reference/ccr/index-following.asciidoc

```diff
@@ -0,0 +1,98 @@
+
+The {ref}/ccr-get-follow-stats.html[get follow stats api] can be used to monitor
+the following process.
+
+=== Index following paused
+
+It is possible to pause index following using the
+{ref}/ccr-post-pause-follow.html[pause following api]. While a follower is
+paused, it will not request new operations from the leader index. The follower
+index is still readable for search requests.
+
+To resume index following, use the
+{ref}/ccr-post-resume-follow.html[resume following api].
+
+=== Leader index retaining operations for replication
+
+If the follower is unable to replicate operations from a leader for a period of
+time, the following process can fail due to the leader lacking a complete history
+of operations necessary for the follower to replicate.
+
+Operations replicated to the follower are identified using a sequence number
+generated when the operation was initially performed. Lucene segment files are
+occasionally merged in order to optimize searches and save space. When these
+merges occur, it is possible for operations associated with deleted or updated
+documents to be pruned during the merge. When the follower requests the sequence
+number for a pruned operation, the process will fail due to the operation missing
+on the leader.
+
+This scenario is not possible in an append-only workflow. As documents are never
+deleted or updated, the underlying operation will not be pruned.
+
+Elasticsearch attempts to mitigate this potential issue for update workflows using
+a Lucene feature called soft deletes. When a document is updated or deleted, the
+underlying operation is retained in the Lucene index for a period of time. This
+period of time is governed by the `index.soft_deletes.retention_lease.period`
+setting which defaults to `12 hours`.
+
+When a follower initiates the index following, it acquires a retention lease from
+the leader. This informs the leader that it should not allow a soft delete to be
+pruned until either the follower indicates that it has received the operation or
+the lease expires after `12 hours`. It is valuable to have monitoring in place to
+detect a follower replication issue prior to `12 hours` passing so that the
+problem can be remedied before the follower falls fatally behind.
+
+=== Remedying a follower that has fallen behind
+
+If a follower falls sufficiently behind a leader that it can no longer replicate
+operations this can be detected using the
+{ref}/ccr-get-follow-stats.html[get follow stats api]. It will be reported as a
+`indices[].fatal_exception`.
+
+In order to restart the follower, the following process must be paused, the follower
+index closed, and the create follower api called again.
+
+["source","js"]
+----------------------------------------------------------------------
+POST /follower-index-name/_ccr/pause_follow
+
+POST /follower-index-name/_close
+
+PUT /follower-index-name/_ccr/follow
+----------------------------------------------------------------------
+// NOTCONSOLE
```

Comment:
Why `// NOTCONSOLE`? It is possible to cheaply set up requests that would actually work here. See for example the resume follower docs. Let us go the extra mile here so that we docs that are tested and therefore can be maintained if we ever make changes to these APIs (for example, it has already been proposed that we rename these to `stop` and `start` 🤦‍♀️).